lib/utils/device/deploy.ts

```diff
@@ -125,6 +124,14 @@ export async function deployToDevice(opts: DeviceDeployOptions): Promise<void> {
 	const { exitWithExpectedError } = await import('../../errors');
 	const { displayDeviceLogs } = await import('./logs');
 
+	// Resolve .local addresses to IP to avoid
+	// issue with Windows and rapid repeat lookups.
+	// see: https://github.com/balena-io/balena-cli/issues/1518
+	if (opts.deviceHost.includes('.local')) {
+		const dns = await import('dns');
+		opts.deviceHost = (await dns.promises.lookup(opts.deviceHost)).address;
```

Comment:
The `.promises` bit -- was this something that didn't work with Node v8? We need to make progress on #1770 and drop Node 8 as soon as we can...

Reply:
```
$ nvm use 8
Now using node v8.17.0 (npm v6.13.4)
$ node
> require('dns').promises
undefined
```
```
$ nvm use 12
Now using node v12.16.3 (npm v6.14.4)
$ node
> require('dns').promises
{ lookup: [Function: lookup],
...
```
Also: I understand that the CLI does not yet support IPv6 (#1585), and looking at the documentation for `dns.lookup(hostname, options)`, it is possible to pass an `options` object to request the IP address resolution to use IPv4 instead of IPv6: `{ family: 4 }`.
https://nodejs.org/docs/latest-v8.x/api/dns.html#dns_dns_lookup_hostname_options_callback
Although it is not strictly necessary "for now" because the documentation also reads:

> verbatim <boolean> When true, the callback receives IPv4 and IPv6 addresses in the order the DNS resolver returned them. When false, IPv4 addresses are placed before IPv6 addresses. Default: currently false (addresses are reordered) **but this is expected to change in the not too distant future.** New code should use { verbatim: true }.

Reply:
Ah yes, I think you're right.  In that case we will need to hold off merging this until after 1770 then I guess.

Reply:
Re ip4, right that might avoid some surprises in the future.

Reply:
> we will need to hold off merging this until after 1770

`dns.promises` is not available but `util.promisify()` can be used instead (with Node v8):

> https://nodejs.org/docs/latest-v8.x/api/dns.html#dns_dns_lookup_hostname_options_callback
> If this method is invoked as its util.promisify()ed version, and all is not set to true, it returns a Promise for an object with address and family properties.